From b552a86d494c1acf9c3472b8e5f00d90f9d3a704 Mon Sep 17 00:00:00 2001
From: Daniel Richard G <iskunk@gmail.com>
Date: Thu, 10 Apr 2025 04:35:40 -0400
Subject: [PATCH] Don't apply /*FALLTHROUGH*/ edit to gperf 3.2 output

The gperf issue at https://savannah.gnu.org/bugs/index.php?53029
has been resolved as of the 3.2 release, and not only is the
/*FALLTHROUGH*/ comment replacement no longer needed, it now
breaks the build with "error: fallthrough annotation does not
directly precede switch label". Only do the edit for 3.1.

Bug: None
Change-Id: I1f12a2a685b62d0dedb4298bc171ab4c94ec6b47
---

diff --git a/third_party/blink/renderer/build/scripts/gperf.py b/third_party/blink/renderer/build/scripts/gperf.py
index 42630d3..0df5abd 100644
--- a/third_party/blink/renderer/build/scripts/gperf.py
+++ b/third_party/blink/renderer/build/scripts/gperf.py
@@ -35,10 +35,11 @@
         # https://savannah.gnu.org/bugs/index.php?53028
         gperf_output = re.sub(r'\bregister ', '', gperf_output)
         # -Wimplicit-fallthrough needs an explicit fallthrough statement,
-        # so replace gperf's /*FALLTHROUGH*/ comment with the statement.
-        # https://savannah.gnu.org/bugs/index.php?53029
-        gperf_output = gperf_output.replace('/*FALLTHROUGH*/',
-                                            '  [[fallthrough]];')
+        # so replace gperf 3.1's /*FALLTHROUGH*/ comment with the statement.
+        # https://savannah.gnu.org/bugs/index.php?53029 (fixed in 3.2)
+        if '/* C++ code produced by gperf version 3.1 */' in gperf_output:
+            gperf_output = gperf_output.replace('/*FALLTHROUGH*/',
+                                                '  [[fallthrough]];')
         # -Wpointer-to-int-cast warns about casting pointers to smaller ints
         # Replace {(int)(long)&(foo), bar} with
         # {static_cast<int>(reinterpret_cast<uintptr_t>(&(foo)), bar}

